### PR TITLE
vm.args support for node_package

### PR DESCRIPTION
**Pull request by joedevivo**
So, the output in a cuttlefish world looks like this:

```
➜  riak git:(jd-cuttlefish) ✗ ./rel/riak/bin/riak console
egrep: /Users/joe/dev/basho/riak/rel/riak/bin/../etc/vm.args: No such file or directory
grep: /Users/joe/dev/basho/riak/rel/riak/bin/../etc/vm.args: No such file or directory
Node 'riak@127.0.0.1' not responding to pings.
11:56:27.914 [info] Application lager started on node nonode@nohost
11:56:27.914 [info] No app.config detected in /Users/joe/dev/basho/riak/rel/riak/bin/../etc, activating cuttlefish
config is OK
-config /Users/joe/dev/basho/riak/rel/riak/bin/../etc/generated/app.2013.08.29.11.56.27.config -args_file /Users/joe/dev/basho/riak/rel/riak/bin/../etc/generated/vm.2013.08.29.11.56.27.args
Exec: /Users/joe/dev/basho/riak/rel/riak/bin/../erts-5.9.3.1/bin/erlexec -boot /Users/joe/dev/basho/riak/rel/riak/bin/../releases/1.4.0/riak               -config /Users/joe/dev/basho/riak/rel/riak/bin/../etc/generated/app.2013.08.29.11.56.27.config -args_file /Users/joe/dev/basho/riak/rel/riak/bin/../etc/generated/vm.2013.08.29.11.56.27.args              -pa /Users/joe/dev/basho/riak/rel/riak/bin/../lib/basho-patches -- console
Root: /Users/joe/dev/basho/riak/rel/riak/bin/..
Erlang R15B03 (erts-5.9.3.1) [source] [64-bit] [smp:4:4] [async-threads:64] [kernel-poll:true]

Eshell V5.9.3.1  (abort with ^G)
(riak@127.0.0.1)1> q().
ok
```

As you can see, there are some egrep errors I'm going to need some help hiding, since those errors aren't actually important. I figure @jaredmorrow can school me on some sh in the process.

Also, some comments in-line where helpful.

### CODE REVIEW

File: priv/base/env.sh [hunk 1]
```diff
@@ -56,8 +56,11 @@ echoerr() { echo "$@" 1>&2; }
 # Extract the target node name from node.args
 NAME_ARG=`egrep '^\-s?name' $RUNNER_ETC_DIR/vm.args`
 if [ -z "$NAME_ARG" ]; then
```
Comment (joedevivo) [inline priv/base/env.sh:58]: nested if. If name's in vm.args, use it. If not, check conf file.

File: priv/base/env.sh [hunk 2]
```diff
@@ -56,8 +56,11 @@ echoerr() { echo "$@" 1>&2; }
...
-    echoerr "vm.args needs to have either -name or -sname parameter."
-    exit 1
+    NAME_ARG="-name `egrep '^nodename\s?=\s?' $RUNNER_ETC_DIR/riak.conf | cut -d = -f 2`"
```
Comment (joedevivo) [inline priv/base/env.sh]: hard coded riak.conf file. By accident. I will fix, but maybe not before you see the review. This filename is already a var understood by rebar.

File: priv/base/env.sh [hunk 3]
```diff
@@ -78,8 +81,11 @@ fi
 # Extract the target cookie
 COOKIE_ARG=`grep '^\-setcookie' $RUNNER_ETC_DIR/vm.args`
 if [ -z "$COOKIE_ARG" ]; then
-    echoerr "vm.args needs to have a -setcookie parameter."
-    exit 1
+    COOKIE_ARG="-setcookie `egrep '^distributed_cookie\s?=\s?' $RUNNER_ETC_DIR/riak.conf | cut -d = -f 2`"
```
Comment (joedevivo) [inline priv/base/env.sh]: same issue.

File: priv/base/env.sh [hunk 4]
```diff
@@ -242,7 +248,7 @@ get_pid() {
 # This guy sets $CONFIG_FILE to the absolute filename to the app.config
 # generated by cuttlefish
 cuttlefish() {
-    CONFIG_FILE=`$ERTS_PATH/cuttlefish -e $RUNNER_ETC_DIR -s $RUNNER_LIB_DIR -c $RUNNER_ETC_DIR/{{cuttlefish_conf}}`
+    CONFIG_ARGS=`$ERTS_PATH/cuttlefish -e $RUNNER_ETC_DIR -s $RUNNER_LIB_DIR -c $RUNNER_ETC_DIR/{{cuttlefish_conf}}`
```
Comment (joedevivo) [inline priv/base/env.sh]: cuttlefish now outputs "-config filename -args_file filename2" so sub these in later.

File: priv/base/nodetool [hunk 5]
```diff
@@ -29,21 +29,9 @@ main(Args) ->
     %% any commands that don't need a running node
     case RestArgs of
         ["chkconfig", File] ->
-            case file:consult(File) of
```
Comment (joedevivo) [inline priv/base/nodetool:32]: refactored into it's own function

File: priv/base/nodetool [hunk 6]
```diff
@@ -29,21 +29,9 @@ main(Args) ->
...
-                               file:format_error({Line, Mod, Term}), "\n"], []),
-                    halt(1);
-                {error, R} ->
-                    io:format(standard_error,
-                              ["Error reading config file: ", 
-                               file:format_error(R), "\n"], []),
-                    halt(1)
-            end;
+            chkconfig(File);
+        ["chkconfig", "-config", File|_] ->
```
Comment (joedevivo) [inline priv/base/nodetool:33]: allowed nodetool to accept the string output by cuttlefish. old functionality still works.

File: priv/base/runner [hunk 7]
```diff
@@ -228,9 +228,8 @@ case "$1" in
         EMU=beam
         PROGNAME=`echo $0 | sed 's/.*\///'`
         CMD="$BINDIR/erlexec -boot $RUNNER_BASE_DIR/releases/$APP_VSN/$RUNNER_SCRIPT \
-             -config $CONFIG_FILE \
-            -pa $RUNNER_PATCH_DIR \
-            -args_file $RUNNER_ETC_DIR/vm.args -- ${1+"$@"}"
+             $CONFIG_ARGS \
```
Comment (joedevivo) [inline priv/base/runner:333]: $CONFIG_ARGS are both -config and -args_file

**Comment by jaredmorrow** [comment]
+1, please squash before merging.

**Pull request merged**